src/mrinufft/operators/interfaces/nudft_numpy.py

```diff
@@ -4,23 +4,41 @@
 
 import numpy as np
 import scipy as sp
-
 from ..base import FourierOperatorCPU
-from mrinufft._utils import proper_trajectory
+from mrinufft._utils import proper_trajectory, get_array_module
 
 
 def get_fourier_matrix(ktraj, shape, dtype=np.complex64, normalize=False):
     """Get the NDFT Fourier Matrix."""
+    module = get_array_module(ktraj)
     ktraj = proper_trajectory(ktraj, normalize="unit")
     n = np.prod(shape)
     ndim = len(shape)
-    matrix = np.zeros((len(ktraj), n), dtype=dtype)
-    r = [np.linspace(-s / 2, s / 2 - 1, s) for s in shape]
-    grid_r = np.reshape(np.meshgrid(*r, indexing="ij"), (ndim, np.prod(shape)))
-    traj_grid = ktraj @ grid_r
-    matrix = np.exp(-2j * np.pi * traj_grid, dtype=dtype)
+    dtype = module.complex64
+    device = getattr(ktraj, "device", None)
+
+    r = [module.linspace(-s / 2, s / 2 - 1, s) for s in shape]
+    if module.__name__ == "torch":
+        r = [x.to(device) for x in r]
+
+    grid_r = module.meshgrid(*r, indexing="ij")
+    grid_r = module.reshape(module.stack(grid_r), (ndim, n))
+    traj_grid = module.matmul(ktraj, grid_r)
+    matrix = (
+        module.exp(-2j * module.pi * traj_grid).to(dtype).to(device).clone()
+        if module.__name__ == "torch"
+        else (module.exp(-2j * module.pi * traj_grid, dtype=dtype))
+    )
+
     if normalize:
-        matrix /= np.sqrt(np.prod(shape)) * np.power(np.sqrt(2), len(shape))
+        norm_factor = (
+            module.sqrt(module.prod(module.tensor(shape, device=device)))
+            * module.pow(module.sqrt(module.tensor(2, device=device)), ndim)
+            if module.__name__ == "torch"
+            else (module.sqrt(module.prod(shape)) * module.power(module.sqrt(2), ndim))
+        )
+        matrix /= norm_factor
```

Comment:
@paquiteau , ideas to have the code cleaner?

Reply:
The ternary is too long, use  a regular if-else block.
Also, the norm factor is just a scalar, everything can be done in numpy regardless of the application afterwards.